CRM/Contact/BAO/Query.php

```diff
@@ -4898,16 +4907,16 @@ public function searchQuery(
     list($select, $from, $where, $having) = $this->query($count, $sortByChar, $groupContacts, $onlyDeleted);
 
     if (!empty($groupByCols)) {
+      if (!empty($order)) {
+        // retrieve order by columns from ORDER BY clause
```

Comment:
and this re-ordering - not sure the reason

Reply:
I suspect the reasoning is that if any of the orderBy Columns are not already in the Select statement then they would be added to the GroupBy but not Select because of L4917